### ISSUE TITLE
Exercise Head Direction

### ISSUE THREAD

**Issue by BalzaniEdoardo** [flatironinstitute/nemos-workshop-feb-2024#7]
Exercise Head Direction

1. Ask them to create the shifted current feature matrix by completing the loop
```python
for i in range(window_size, neuron_count.shape[0]):
    # insert indices to grab the count history
    from_idx= 
    to_idx =
    input_feature[i-window_size, 0, :] = neuron_count[from_idx:to_idx]
```
have some code to visualize the output
2. Once they are satisfied with the feature, instantiate fit a glm ﻿ ﻿
```python
# Define glm
model = nmo.glm.GLM(
# fit model
model.fit(
```
3. ask them to grab and plot weights
```python
weights = model.
plt.plot(weights)
```
4. Ask to describe the weights, how do we interpret it? are the fast fluctuations real? probably not;
5. Compress the feature; ask them to compute the dot product between the feature and the basis, check the dimensionality, provide code that plots the result
6. Fit with the new fearures and compare weights
7. Walk them through the construction of the fully coupled model: ﻿
    a. Convolve all neuron (use nemos utils, provide the name of the function) ->`(n_samples, n_neurons, n_basis)`
    b. Reshape the output so that we have a neuron predicted by all others ->`(n_samples, 1, n_neurons * n_basis)` ﻿
    c. Replicate for all neurons, leave the `np.repeat(features, , axis=)` as a hint ﻿ ﻿ ﻿
    d. ask them to verifie that each neuron have the same features, like testing at least `feature[:,i] == feature[:,j]` for some i,j ﻿
    e. Ask them to fit the fully coupled model and plot the weights. Provide plotting code but ask them to reshape the weights from `(n_neurons, n_neurons * n_basis)` ﻿ to  `(n_neurons, n_neurons, n_basis)` . how do the weight look like, any structure? ﻿ ﻿
8. Open exercise: include the head dir as an explicit regressor. does the heading improve model fit? does the weight  change structure? how? can you speculate on why?
